javaCalc/src/Calculator.java

```diff
@@ -4,324 +4,307 @@
 import java.text.NumberFormat;
 
 import javax.swing.*;
-import java.util.*;
 
 public class Calculator extends JFrame {
-	private double savedValue;
-	private double displayedValue;
-	NumberFormat formatter;
-	int decimal;
-	char operation = 'o';
+	private double _savedValue;
+	private double _displayedValue;
+	private NumberFormat _formatter;
+	private int _decimal;
+	private char _operation = 'o';
 	
-	/*private JButton b0;
-	private JButton b1;
-	private JButton b2;
-	private JButton b3;
-	private JButton b4;
-	private JButton b5;
-	private JButton b6;
-	private JButton b7;
-	private JButton b8;
-	private JButton b9;*/
+	private JButton[] _digits;
+	private JButton _dec;
+	private JButton _neg;
 	
-	private JButton[] digits;
-	private JButton dec;
-	private JButton neg;
+	private JButton _equal;
+	private JButton _plus;
+	private JButton _minus;
+	private JButton _times;
+	private JButton _divide;
+	private JButton _square;
+	private JButton _exponent;
+	private JButton _clear;
+	private JButton _clearentry;
 	
-	private JButton equal;
-	private JButton plus;
-	private JButton minus;
-	private JButton times;
-	private JButton divide;
-	private JButton square;
-	private JButton exponent;
-	private JButton clear;
-	private JButton clearentry;
-	
-	private JTextField display;
+	private JTextField _display;
 	
 	// constructor
 	
 	public Calculator() {
-		savedValue = 0;
-		displayedValue = 0;
-		decimal = 0;
-		formatter = new DecimalFormat("########.########");
+		_savedValue = 0;
+		_displayedValue = 0;
+		_decimal = 0;
+		_formatter = new DecimalFormat("########.########");
+		
+		// Make Buttons;
+		
+		makeButtons();
+		
+		// Implement listeners
+		
+		implementListeners();
+		
+		// Make ButtonPanel
+		
+		JPanel buttonPanel = makeButtonPanel();
+		
+		// TextField
+		
+		JPanel textPanel = makeTextPanel();
+		
+		// Put both panels onto frame
+		
+		JPanel p = new JPanel();
+		p.setLayout(new FlowLayout());
+		p.add(textPanel);
+		p.add(buttonPanel);
+		
+		add(p);
+	}
+	
+	// Display functions
+	
+	public void updateDisplay(){
+		_display.setText(_formatter.format(_displayedValue));
+	}
+	
+	public void resetDisplay(){
+		_displayedValue = 0;
+		_decimal = 0;
+		updateDisplay();
+	}
+	
+	public void allCancel(){
+		_savedValue = 0;
+		_operation = 'o';
+		resetDisplay();
+	}
+	
+	public void clearScreen(){
+		if(_displayedValue == 0) {
+			allCancel();
+		}
+		else resetDisplay();
+	}
+
+	public void saveValue() {
+		_savedValue = _displayedValue;
+		clearScreen();
+	}
+	
+	public void changeOperation(char op) {
+		_operation = op;
+		saveValue();
+	}
+
+	// Initialization Functions
+	
+	public JButton makeButton(String text, int height, int width) {

```

Comment:
you should make this private, or use the init class to add this functionality to the calc class